src/sql/workbench/contrib/dashboard/browser/core/dashboardPage.component.ts

```diff
@@ -126,6 +151,48 @@ export abstract class DashboardPage extends AngularDisposable implements IConfig
 			};
 			this.createTabs(tempWidgets);
 		}
+
+		this.createTaskbar(this.taskContainer.nativeElement);
+	}
+
+	protected createTaskbar(parentElement: HTMLElement): void {
+		let taskbarContainer = DOM.append(parentElement, DOM.$('div'));
+		this.taskbar = this._register(new Taskbar(taskbarContainer));
+		this._restoreAction = new RestoreToolbarAction(this.restore, this);
+		this._newQueryAction = new NewQueryAction(this.newQuery, this);
+		this._newNotebookAction = new NewNotebookToolbarAction(this.newNotebook, this);
+		this._editAction = new EditDashboardAction(this.enableEdit, this);
+		this._refreshAction = new RefreshWidgetAction(this.refresh, this);
+		this._manageExtensionsAction = new ManageExtensionsToolbarAction(this.manageExtensions, this);
+		this.setTaskbarContent();
+	}
+
+	protected setTaskbarContent(): void {
+		// Create HTML Elements for the taskbar
+		let separator: HTMLElement = Taskbar.createTaskbarSeparator();
+		separator.style.margin = '0px 15px 0px 0px';
```

Comment:
Why isn't this done in the createTaskbarSeparator? Having random values like this floating around make it difficult to track down if there's issues.

And anyways this should probably go into a CSS anyways